Upgrade to Angular 9 - fix issues #45 and #36

blavenie opened the pull request.

blavenie committed:
Merge remote-tracking branch 'remote_origin/master'

blavenie committed:
Upgrade to Angular 9 - fix #45

blavenie committed:
Upgrade dev deps, to Angular 9 - fix #45

blavenie committed:
Fix compilation (use ngcc)

blavenie committed:
- CHange tsonfig, to target es2015 (instead of es5) - fix #36
- Upgrade dev deps (replace uglify-js with ulgify-es@3.3.10)
- Add rollup plugins as dev deps
- Fix rollup.config, to use plugins

blavenie commented:
Hi @irossimoline : If you find some time, to check my patch, i'll appreciate :) 

Did you have a BTC wallet, for a tip ? or better: a [libre currency G1](https://duniter.org/en/introduction/) wallet ? You create one from with wallet tool [cesium.app](https://cesium.app). I would like to give you some money for your work.

blavenie committed:
Upgrade to @angular/cli@9.1.6

blavenie committed:
Fix build error "could not be resolved to an NgModule class" on angular/commons

blavenie commented:
This time, everything should run (`npm install` and `npm run build`).

blavenie committed:
Upgrade to Angular 10

blavenie committed:
Remove dependency to 'rxjs/internal/operators'

blavenie committed:
- Update Angular libs
- Change project name to ngx-material-table

blavenie committed:
Fix Git URL

blavenie committed:
Chnage project name to '@e-is/ngx-material-table'

blavenie committed:
Fix rollup build

blavenie committed:
[enh] Add example page
[enh] Build using ng-package and packagr

blavenie committed:
Fix build (bad path in public_api.ts)

blavenie committed:
Create release v0.10.2

irossimoline commented:
Hi there @blavenie, sorry for the (huge) delay! 
I have been very busy lately, but I see you have done some amazing work here. 🎉 

I'll try to reach you through linkedin to talk a bit about it. 😄

blavenie commented:
Hi ! I also changed build scripts, to use `ng-packagr` : It build better artifacts, compatible with main ECMA script versions.
I also rename the project, to be able to release it without wayting you (see https://github.com/e-is/ngx-material-table)

Let me know if you think i can help again.

(I do not use linkedin anymore - i try to go away from US GAFAM / big companies :) )

blavenie committed:
Create release v0.11.0, for Angular 11

blavenie committed:
Create release v0.11.0, for Angular 11

blavenie committed:
Create release v0.11.0, for Angular 11

blavenie committed:
Create release v0.11.0, for Angular 11

blavenie committed:
Create release v0.11.1, using ng-packagr 11

blavenie committed:
Create release v0.11.1, using ng-packagr 11

blavenie committed:
Add publish config

LudovicPecquot committed:
[enh] table-element.ts: add option restoreOriginalData on cancelOrDelete to restore original data instead of previous one

LudovicPecquot committed:
[enh] table-element.ts: remove option restoreOriginalData on cancelOrDelete, prefer option on TableDataSource constructor

LudovicPecquot committed:
update doc

LudovicPecquot committed:
[enh] add TableDataSourceOptions

LudovicPecquot committed:
[enh] remove duplicated option

LudovicPecquot committed:
[enh] update package version

LudovicPecquot committed:
[fix] don't set originalData at row instantiation
[enh] if keepOriginalDataAfterConfirm is true, on confirm row creation, save current data to original data

LudovicPecquot committed:
[enh] table-data-source.ts: createNew accepts insertAt parameter to add new row at specific position (todo need test)
[enh] table-data-source.ts: add method move() to move a row up or down the table

LudovicPecquot committed:
[fix] update row ids on row moved

LudovicPecquot committed:
[fix] table-element-reactive-forms.ts: isValid() return real validator.valid status

LudovicPecquot committed:
[fix] table-data-source.ts: better handling of new row when inserted

LudovicPecquot committed:
release 0.11.9

blavenie committed:
Update publish config

blavenie committed:
[fix] Fix publish scripts

blavenie committed:
[enh] Delegate all row actions to source

blavenie committed:
[enh] Delegate all row actions to source

blavenie committed:
Release 0.12.0-rc6

blavenie committed:
Release 0.12.0

blavenie committed:
[enh] add property row.dirty

Commit:
Release 0.14.0-rc4
[enh] Upgrade to Angular 14 (without all functions as async)

Commit:
Release 0.14.0
[enh] Add AsyncTableDataSource to manage async validator

Commit:
Release 0.14.1
[enh] Add AsyncTableElement
Release 0.14.2
[fix] Revert to rxjs 6.6.7

Commit:
[fix] Revert to rxjs 6.6.7

Commit:
Release 0.14.3

blavenie committed:
Release 0.14.4
[fix] TableDataSource: when opts.editing is undefined, use true by default

blavenie committed:
Release 0.14.5
[fix] Fix row id update, when move/insert preprend rows
[enh] Add TableDatasource.addMany() to insert many rows
[enh] Add TableDatasource.hasNewElement: boolean to known if there is a new+editing row
[enh] Add examples, using nav tabs
[enh] Add ScrollableTableDataSource to use CDK virtual scrolling (expérimental - not expose yet)

blavenie committed:
[enh] Add changelog

blavenie committed:
[fix] Merge from master (for 0.12.1)

blavenie committed:
Release 0.14.6

blavenie committed:
Release 0.14.7
Make sure to disable the row validator, even if seem to be already disabled

blavenie committed:
Release 0.14.7
Make sure to disable the row validator, even if seem to be already disabled

blavenie committed:
Release 0.14.8
Revert changes from 0.14.7

blavenie committed:
Release 0.14.9
Minor changes

blavenie committed:
Release 0.14.9
Minor changes

blavenie committed:
Remove a console log

blavenie committed:
Upgrade angular libs

blavenie committed:
Upgrade angular 15

blavenie committed:
Upgrade angular material 15

blavenie committed:
Release v0.15.0

LudovicPecquot committed:
release 0.14.11
[fix] add unwrapESModule

LudovicPecquot committed:
release 0.14.12
[enh] upgrade Angular to 14.3.0

LudovicPecquot committed:
upgrade libs (using legacy angular material)

indelog committed:
Merge remote-tracking branch 'origin/upgrade_angular' into v0.15.x

# Conflicts:
#	package.json
#	src/styles.scss

indelog committed:
Upgrade angular 16

indelog committed:
Upgrade angular 17

LudovicPecquot committed:
upgrade angular libs

LudovicPecquot committed:
upgrade libs and remove zonejs

LudovicPecquot committed:
Release 17.0.0

blavenie committed:
enh(deps) upgrade to Angular 17.2.1

blavenie committed:
enh(deps) upgrade to Angular 17.2.1
enh(prettier) Run prettier --write and eslint --fix

blavenie committed:
enh(deps) upgrade to Angular 17.2.1
enh(prettier) Run prettier --write and eslint --fix

blavenie committed:
Merge remote-tracking branch 'origin/master'

blavenie committed:
fix(scrollable) Remove duplicated class AsyncTableElement
enh(table-element) Add options to `row.delete()`, `row.confirmEditCreate()` and `row.cancelOrDelete()`

blavenie committed:
Commit previous 17.2.0

blavenie committed:
Release 17.2.1

blavenie committed:
fix compilation error, by making ng-table-virtual-scroll optional

blavenie committed:
fix compilation error, by making ng-table-virtual-scroll optional

blavenie committed:
Release 17.2.2

blavenie committed:
Release 17.3.0

blavenie committed:
Release 17.3.0

blavenie committed:
wip(migration) Start Angular 18 migration

blavenie committed:
enh(deps) Migrate to Angular 18

blavenie committed:
enh(deps) Migrate to Angular 18

blavenie committed:
[fix] `DataSource.createNew()` should always use the value of `insertAt` when equals to 0 (change `if (insertAt) { /*...*/ }` into `if (insertAt >= 0) { /*...*/ }`)

blavenie committed:
[fix] `DataSource.createNew()` should always use the value of `insertAt` when equals to 0 (replace `if (insertAt) { /*...*/ }` with `if (insertAt >= 0) { /*...*/ }`)

blavenie committed:
Release 18.0.2

blavenie committed:
[fix] Fix error when calling `DataSource.addMany()` with `insertAt=null`

blavenie committed:
Release 18.0.4 - Fix webpack version 5.94.0

blavenie committed:
Update package description and keywords for Angular 18 compatibility

blavenie committed:
Update dependencies and engine requirements for Angular 18 compatibility

blavenie committed:
enh(deps) Migrate to Angular 19

blavenie committed:
enh(deps) Migrate to Angular 20

blavenie committed:
enh(deps) Migrate to Angular 20

blavenie committed:
docs: update README and dependencies for Angular 20 compatibility
